### DISCUSSION TITLE
Multi-selection of elements in stencil

### DISCUSSION THREAD

**Discussion by kvinita** [clientIO/joint#2605] [Q&A]
### Introduction

I have elements in the stencil and I wish to select multiple elements from stencil at once and drag them to paper. Is such multiselect operation possible in stencil?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
It can be implemented using the current stencil API (and events).
- Every group in a stencil is just a `dia.Paper` (`stencil.getPaper(groupId)`). That means the `ui.Selection` or `highlighters` can be used.
- You can display a custom element during the dragging such an element that represents a group of elements (`dragStartClone` option)
- You can cancel the default drag and add as many new elements to the graph as you desired (on 'element:dragend')

Here's a simple example using custom highlighters: https://codepen.io/jointjs/pen/abQyLew
Similar issue: https://github.com/clientIO/joint/discussions/2546

**Reply by kvinita** [reply]
My requirement is that, I should be able to select any number of elements in the stencil at a time using Ctrl functions. It's dynamic. Hence, I think grouping the stencil elements beforehand is not right. Am I right?

**Reply by kumilingus** [reply]
It's ok to use stencil groups. You just need to check the selected elements across all groups.
